### DISCUSSION TITLE
[Feature] Immich uploader should deprioritize failed uploads

### DISCUSSION THREAD

**Discussion by dfcowell** [immich-app/immich#2892] [Feature Request]
### The feature

This is probably less of an issue if chunked uploads get implemented, however until then...

I have a single large video on my device. Immich's upload logic appears to be reverse-chronological, meaning that every time a backup starts, it tries to upload this video, even though it will never work. Due to the large file size it takes a substantial amount of time.

The Immich client app should put files that have previously failed to upload at the end of the upload queue to unblock uploading other files that may upload successfully. This should substantially speed up the process of backing up a large library that contains a couple of large files.

### Platform

- [X] Server
- [ ] Web
- [X] Mobile

**Comment by mmomjian** [comment]
#6173
